packages/core/src/metadata/EntitySchema.ts

```diff
@@ -22,7 +22,7 @@ type Property<T, O> =
 type PropertyKey<T, U> = NonFunctionPropertyNames<Omit<T, keyof U>>;
 type Metadata<T, U> =
   & Omit<Partial<EntityMetadata<T>>, 'name' | 'properties'>
-  & ({ name: string } | { class: Constructor<T>; name?: string })
+  & ({ name: string } | { class: Constructor<T>; name?: string } | { name: string; schema: string })
```

Comment:
this looks wrong, I'd say it should be this instead (otherwise you'd not allow using `class` with `schema`)
```suggestion
  & ({ name: string; schema?: string } | { class: Constructor<T>; name?: string; schema?: string })
```

Reply:
Eh? I think I got some TS error somewhere in this regards. Maybe I added it here before I added it to `EntityMetadata<>`.